packages/next/src/server/stream-utils/node-web-streams-helper.ts

```diff
@@ -93,8 +90,16 @@ export function createBufferedTransformStream(): TransformStream<
 
     scheduleImmediate(() => {
       try {
-        controller.enqueue(buffer)
-        buffer = new Uint8Array()
+        const chunk = new Uint8Array(bufferSize)
+        let copiedBytes = 0
+        for (let i = 0; i < bufferedChunks.length; i++) {
+          const bufferedChunk = bufferedChunks[i]
+          chunk.set(bufferedChunk, copiedBytes)
+          copiedBytes += bufferedChunk.byteLength
+        }
+        bufferedChunks.length = 0
```

Comment:
I understand that this is clearing the array, but it would be good to comment that this is what we're doing here.